File: Pilha/Pilha.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Pilha {
	int topo;
	int itens[5];
} Pilha;

void inicializar(Pilha *p) {
	p -> topo = -1;
//	(*p).topo = -1;	
}

int mostrarMenu() {
	int opc;
	
	printf("--------------- MENU ---------------\n\n");
	printf("[1] - Inserir um elemento na Pilha\n");
	printf("[2] - Remover um elemento da Pilha\n");
	printf("[3] - Verificar condicao da Pilha\n");
	printf("[0] - Encerrar o Programa\n\n");
	printf("------------------------------------\n\n");
	printf("Digite sua escolha: ");
	scanf("%d",&opc);
	system("cls");
	
	return opc;
}

void inserirPilha(int e, Pilha *p) {

	if (p -> topo == 4) {
		printf("\n!!! A PILHA ESTA CHEIA !!!\n\nNao eh possivel inserir mais elementos.");
	}
	
	else {
		p -> topo++;
		p -> itens[p -> topo] = e;
        printf("\nElemento %d foi inserido na pilha.", e);
	}
}

void removerPilha(Pilha *p) {
	int elementoRemovido;
	
	if (p -> topo == -1) {
		printf("!!! A PILHA ESTA VAZIA !!!\n\nNao eh possivel remover mais elementos.");
	}
	
	else {
		elementoRemovido = p -> itens[p -> topo];
		p -> topo--;
		printf ("O elemento %d foi removido da pilha.", elementoRemovido);
	}
}

void verificarCondicaoPilha(Pilha *p) {
	int i;
	
	if (p -> topo == 4) {
		printf("!!! A PILHA ESTA CHEIA !!!\n\n");
	}
	
	else if (p -> topo == -1) {
		printf("!!! A PILHA ESTA VAZIA !!!");
	}
	
	if (p -> topo != -1) {
		printf ("A sua pilha atual eh: \n\n");
		printf ("PILHA = ");
		for(i = 0; i <= p -> topo; i++) {
			printf ("- [ %d ] ", p -> itens[i]);
		}
	}
}


int main() {
	int opc;   //Variavel que atribuira a opcao do usuario
	int elemento;   //Numero que vai ser inserido na pilha
	Pilha p;

	inicializar(&p);
	
	do {
		opc = mostrarMenu();
		
		if (opc >= 4 || opc < 0) {
			printf("OPCAO INVALIDA - Escolha uma nova opcao. \n\n");
		}
		
		if (opc == 1) {
			printf("--------------- ENTRADA ---------------\n\n");
			printf("Digite um elemento para entrar na pilha: ");
			scanf("%d",&elemento);

			inserirPilha(elemento, &p);
            
			printf("\n\n");
			printf("---------------------------------------\n\n");
			system("pause");
			system("cls");
		}
		
		if (opc == 2) {
			printf("--------------- REMOCAO ---------------\n\n");

			removerPilha(&p);

			printf("\n\n");
			printf("---------------------------------------\n\n");
			system("pause");
			system("cls");	
		}
		
		if( opc == 3) {
			printf("--------------- CONDICAO ---------------\n\n");

			verificarCondicaoPilha(&p);

			printf("\n\n");
			printf("----------------------------------------\n\n");
			system("pause");
			system("cls");
		}
		
		if (opc == 0) {
			printf("--------------- SAIDA ---------------\n\n");
	        printf("Encerrando o Programa...\n\n");
            printf("-------------------------------------\n\n");

            system("pause");
		}
	
	} while (opc != 0);
	
	return 0;
}

File: Pilha/Pilha.py
class Pilha:
    def __init__(self):
        self.topo = -1
        self.itens = [0] * 5

def mostrar_menu():
    print("------------------- MENU -------------------\n")
    print("[1] - Inserir um elemento na Pilha")
    print("[2] - Remover um elemento da Pilha")
    print("[3] - Verificar condição da Pilha")
    print("[0] - Encerrar o Programa\n")
    print("--------------------------------------------\n")
    opc = int(input("DIGITE SUA ESCOLHA: "))
    print()
    return opc

def inserir_pilha(e, p):
    p.topo += 1
    p.itens[p.topo] = e
    print("\nElemento", e, "foi inserido na pilha.\n\n\n")

def remover_pilha(p):
    if p.topo == -1:
        print("!!! A PILHA ESTA VAZIA !!!\n")
        print("Não é possível remover mais elementos.\n\n\n")
    else:
        elemento_removido = p.itens[p.topo]
        p.topo -= 1
        print("O elemento", elemento_removido, "foi removido da pilha.\n\n\n")

def verificar_condicao_pilha(p):
    if p.topo == 4:
        print("!!! A PILHA ESTA CHEIA !!!\n")
        print("PILHA =", end=" ")
        for i in range(p.topo + 1):
            print("- [", p.itens[i], "]", end=" ")
        print("\n\n\n")
    elif p.topo == -1:
        print("!!! A PILHA ESTA VAZIA !!!\n\n\n")
    elif p.topo != -1:
        print("A sua pilha atual é:\n")
        print("PILHA =", end=" ")
        for i in range(p.topo + 1):
            print("- [", p.itens[i], "]", end=" ")
        print("\n\n\n")

p = Pilha()

while True:
    opc = mostrar_menu()
    
    if opc >= 4 or opc < 0:
        print("OPCAO INVALIDA - Escolha uma nova opcao.\n\n")
    
    if opc == 1:
        print("----------------- ENTRADA -----------------\n")
        if p.topo != 4:
            elemento = int(input("Digite um elemento para entrar na pilha: "))
            inserir_pilha(elemento, p)
        else:
            print("!!! A PILHA ESTA CHEIA !!!\n\nNão é possível inserir mais elementos.\n\n\n")

    if opc == 2:
        print("----------------- REMOCAO -----------------\n")
        
        remover_pilha(p)
    
    if opc == 3:
        print("----------------- CONDICAO -----------------\n")
        
        verificar_condicao_pilha(p)
    
    if opc == 0:
        print("------------------ SAIDA ------------------\n")
        print("Encerrando o Programa...            =)\n")
        input("-------------------------------------------\n\n")
        break
